streams/src/main/java/org/apache/kafka/streams/state/OffsetCheckpoint.java

```diff
@@ -55,7 +56,6 @@
     private final Object lock;
 
     public OffsetCheckpoint(File file) throws IOException {
-        new File(file + ".tmp").delete(); // try to delete any existing temp files for cleanliness
```

Comment:
I realized while reviewing this that the problem that 836cb196/KAFKA-3063 attempts to fix may not entirely be fixed. Removing the deletion avoids cases where constructing the object just to read the file causes the data to be deleted.

But isn't there just a more general concurrency issue? If the deletion was causing a problem, doesn't that mean multiple tests were using the same path the files, and the same paths to the temp files? Which means there can, more generally, be errors due to concurrent use of those files?

Wouldn't a complete fix for the tests require that we make sure the filenames never conflict in the first place?